src/Format.js

```diff
@@ -98,7 +110,7 @@ export default class Format {
 			if (!type) {
 				// Type does not exist in the grammar, throw
 				let coordName = coordMeta.name || id;
-				throw new TypeError(`${ providedType ?? arg?.raw ?? arg } not allowed for ${coordName} in ${this.name}()`);
+				throw new TypeError(`${ providedType ?? /** @type {any} */ (arg)?.raw ?? arg } not allowed for ${coordName} in ${this.name}()`);
```

Comment:
Excuse my ignorance, why do we need inline `@type` comments? Is this information we cannot capture in the block jsdoc comments? (a link is fine, I just googled it and couldn't figure it out)

Reply:
This is how type assertions work when using JSDoc.
Type assertions: <https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#type-assertions>
JSDoc syntax for assertions: <https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html#casts>

It's needed whenever TypeScript can't properly determine the type of something within the body of a function (eg. for variables that were defined within that function).

For the line you commented on, TypeScript would complain that the property `raw` doesn't exist on `number`. The easiest way to stop it from complaining was to assert `arg` as `any`, effectively disabling type checking for `arg` on that particular line.

It's also sometimes needed when a function's return type isn't specific enough, like here:

<https://github.com/color-js/color.js/blob/9661dc697426a3983e7d390daea5e49e80291f05/src/RGBColorSpace.js#L42>

Reply:
Ah, I see, thank you!